xgi/utils/tensor.py

```diff
@@ -0,0 +1,191 @@
+## Tensor times same vector in all but one (TTSV1) and all but two (TTSV2)
+from collections import defaultdict
+from itertools import combinations
+from math import factorial
+
+import numpy as np
+from numpy import prod
+from scipy.signal import convolve
+from scipy.sparse import coo_array
+from scipy.special import binom as binomial
+
+__all__ = [
+    "pairwise_incidence",
+    "ttsv1",
+    "ttsv2",
+]
+
+
+def pairwise_incidence(edgedict):
+    """Create pairwise adjacency dictionary from hyperedge list dictionary
+
+    Parameters
+    ----------
+    edgedict : dict
+        edge IDs are keys, edges are values
+
+    Returns
+    -------
+    pairs : dict
+        a dictionary with node pairs as keys and the hyperedges they appear in as values
+    """
+    pairs = defaultdict(set)
+    for e, edge in edgedict.items():
+        for i, j in combinations(sorted(edge), 2):
+            pairs[(i, j)].add(e)
+        for n in edge:
+            pairs[(n, n)].add(e)
+    return pairs
+
+
+def banerjee_coeff(l, r):
+    """Return the Banerjee alpha coefficient
+
+    Parameters
+    ----------
+    l : int
+        size of given hyperedge
+    r : int
```

Comment:
should we replace these by "s" or "size" and "s_max" to make it more explicit and consistent with the rest of the library?

Reply:
I'm down with "size"!

Reply:
Fixed.